modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js

```diff
@@ -75,8 +75,20 @@ AUI.add(
 				},
 
 				rowClassNameActive: {
-					validator: Lang.isString,
-					value: 'active table-active',
+					setter(rowClassNameActive) {
+						if (Lang.isString(rowClassNameActive)) {
+							var active = rowClassNameActive.split(',');
+							var rowSelector = [
+								'li[data-selectable="true"]',
+								'tr[data-selectable="true"]',
+							];
+
+							return active[
+								rowSelector.indexOf(this.get(STR_ROW_SELECTOR))
+							];
+						}
```

Comment:
I must confess that I don't actually understand how this works. Let me try to put it into words and you can confirm for me whether I've understood it:

-   Previously, the value here was `active table-active`; now it is `active,table-active`.
-   There is no reference to `rowClassNameActive` outside of this file, so just based on what I see in here, the value is going to get passed to into `addClass` or `toggleClass`. I can't find the docs or the implementation for YUI/AUI `addClass`, but at least [this line](https://github.com/liferay/alloy-ui/blob/c696a0c0fcf17deb2302dd9dcd913af88bd5c221/src/aui-node/js/aui-node-base.js#L628) suggests to me that it expects a space-separated list of class names; I am not sure what it will do when given a comma-separated list...
-   I don't know AUI/YUI, so I don't know if this `setter` is magically called somehow; I can't see any setter calls anywhere (and like I said, can't see any references outside of this file). My _guess_ here is that your intent is that your `split()` call turns the invalid comma-containing "name" into a single name, but I have no idea _when_ the setter is called.
-   There's obviously a relationship between the classes in `rowSelector` below and the ones on line 81 above, but it seems subtle and fragile, and the whole thing seems roughly equivalent to this:

    ```js
    return 'active,table-active'.split(',')[
        [
            'li[data-selectable="true"]',
            'tr[data-selectable="true"]',
        ].indexOf('li[data-selectable="true"],tr[data-selectable="true"]'),
    ];
    ```

    I don't really get it, as that `indexOf()` looks like it is always going to return `-1`, so the access by array-index of `-1` into the array will always return `undefined`. Maybe somebody who knows AUI a bit better can review this? cc @jonmak08

Reply:
> I must confess that I don't actually understand how this works.

You and me both. This is always magic to me.

> I don't know if this setter is magically called somehow; 

I think setter gets called via `aui-component.js` and `widget-base.js` (YUI), maybe these two files will clear up when it gets called? :shrug:

Reply:
Well, I don't think I can forward this without understanding how it works, especially if you also say that _you_ don't understand it. 😬 Can you think of anybody who would be better qualified to review it?

Reply:
![good-point](https://user-images.githubusercontent.com/788266/108877729-5d9c0080-75b4-11eb-9aa4-6936f675ceac.gif)

@jonmak08 could review it but had a power outage yesterday, otherwise @jbalsas haha.

Reply:
To be honest, I don't know how or why would this work either :joy: @pat270, how did you come up with this script then? Trial and error?

Our main logic is in this same file some lines below:

```javascript
row.toggleClass(instance.get(STR_ROW_CLASS_NAME_ACTIVE));
```

Where `row` is the `li` for cards and descriptive visualizations and the `tr` for tables. So... 

> Card View image card doesn't show active border because it is a special case where we can't style the card with the input:checked ~ .card selector or with the active class on the li element.

I'm not sure I understand what's the technical limitation here... 🤔

Reply:
I would say 65% trial and error 😄 . When we use `active table-active` or `active,table-active` in `rowClassNameActive`, `instance.get(STR_ROW_CLASS_NAME_ACTIVE)` just outputs the string as is (e.g., `class="active,table-active"`). I wasn't able to pinpoint where `instance.get(STR_ROW_SELECTOR)` is able to get the correct selector from `'li[data-selectable="true"],tr[data-selectable="true"]'` so I couldn't copy it.

I saw we could process values using the `setter` method from:  https://github.com/liferay-frontend/liferay-portal/blob/d39915e732be15721347481da00170fbdc4e9296/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js#L57 and did what I needed to do.

I'm still stumped on why `instance.get(STR_ROW_SELECTOR)` can return the correct value from a string separated by commas.

Reply:
Ok I dug around a little...`rowSelector` gets updated by https://github.com/liferay/liferay-portal/blame/master/portal-web/docroot/html/taglib/ui/search_iterator/lexicon/bottom.jspf#L55-L68. I think we could pass in `rowClassNameActive` there and revert the changes made in `search_container_select.js`. Let me know if this is the way.

Reply:
> Ok I dug around a little...rowSelector gets updated by https://github.com/liferay/liferay-portal/blame/master/portal-web/docroot/html/taglib/ui/search_iterator/lexicon/bottom.jspf#L55-L68. I think we could pass in rowClassNameActive there and revert the changes made in search_container_select.js. Let me know if this is the way.

What would we achieve with that? It's still unclear to me what's the current state and what's the final state we'd want it to be. Could you detail the current and expected markup for the active/inactive case and a little explanation as to what has to be that way?

> Card View image card doesn't show active border because it is a special case where we can't style the card with the input:checked ~ .card selector or with the active class on the li element.

This is still my main doubt. Why can't we simply add the `active` card on the `li` and style it with that?

Reply:
I suppose we had a reason to not use `active` and we changed to `table-active`. If not, probably changing the class name could make it easier

Reply:
> did what I needed to do.

![what-i-had-to-do](https://user-images.githubusercontent.com/7074/109014913-e13d1680-76b4-11eb-9344-faa09ffab9dd.gif)

Reply:
So... shot in the dark here:

Is the goal of this PR to simply have the `active` class in the `li` and not both `active` and `table-active`?

Reply:
There are two issues:

1) https://issues.liferay.com/browse/LPS-127450 (Styles for active row is lost). Bootstrap 4 changed the `active` class on table rows to `table-active`. The compat layer removal exposes this issue on search container table view.

2) https://issues.liferay.com/browse/LPS-127831 (Extra styles are added to selected card). The fix from LPS-127450 caused this. The light blue border/background around the card (caused by `table-active`) is what needs to be fixed.

![table-active-on-card](https://user-images.githubusercontent.com/788266/109024885-0c067b00-7673-11eb-97de-55bb39983d45.jpg)

The Clay pr will add the blue border around image card when the card is checked.

Reply:
Got it now, thanks @pat270!

Reply:
Thanks for sending the update @jbalsas!